From bunk@stusta.de Sun Jul 30 16:44:24 2006
Date: Mon, 31 Jul 2006 01:43:53 +0200
From: Adrian Bunk <bunk@stusta.de>
To: gregkh@suse.de
Cc: <toralf.foerster@gmx.de>
Subject: USB: fix the USB_GADGET_DUMMY_HCD dependencies
Message-ID: <20060730234353.GB3658@stusta.de>
Content-Disposition: inline

If USB=m, USB_GADGET=y, the option USB_GADGET_DUMMY_HCD mustn't be 
offered since selecting it results in a compile error.

This patch fixes kernel Bugzilla #6534 reported by Toralf Förster.

Signed-off-by: Adrian Bunk <bunk@stusta.de>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/gadget/Kconfig |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- gregkh-2.6.orig/drivers/usb/gadget/Kconfig
+++ gregkh-2.6/drivers/usb/gadget/Kconfig
@@ -207,7 +207,7 @@ config USB_AT91
 
 config USB_GADGET_DUMMY_HCD
 	boolean "Dummy HCD (DEVELOPMENT)"
-	depends on USB && EXPERIMENTAL
+	depends on (USB=y || (USB=m && USB_GADGET=m)) && EXPERIMENTAL
 	select USB_GADGET_DUALSPEED
 	help
 	  This host controller driver emulates USB, looping all data transfer
